applications/nrf5340_audio/src/audio/streamctrl.c

```diff
@@ -170,6 +170,24 @@ static void le_audio_rx_data_handler(uint8_t const *const p_data, size_t data_si
 	ret = data_fifo_pointer_first_vacant_get(&ble_fifo_rx, (void *)&iso_received, K_NO_WAIT);
 	ERR_CHK_MSG(ret, "Unable to get FIFO pointer");
 
+	uint32_t bitrate, sampling_rate, octets_per_frame;
+	static uint32_t data_size_mismatch_cnt;
+
+	if (CONFIG_AUDIO_DEV == HEADSET) {
+		ret = le_audio_config_get(&bitrate, &sampling_rate, &octets_per_frame);
+		if (ret) {
+			LOG_ERR("Failed to get configs");
+		}
+
+		if (data_size != octets_per_frame) {
+			data_size_mismatch_cnt++;
+			if ((data_size_mismatch_cnt % 500U) == 0U) {
```

Comment:
Great. You need to print before the increment though. If not, we get 499 errors before we see anything :)

Reply:
Or rather, check that the modulo remainder is 1